internal/clients/telegram/conversation_state.go

```diff
@@ -29,35 +31,106 @@ type ConversationStateHandler interface {
 // The default state
 type rootConversationState struct{}
 
+func (*rootConversationState) String() string {
+	return "rootState"
+}
+
 func (s *rootConversationState) telegramMessage(message message) (ConversationStateHandler, []telegramBotActor) {
-	log.Printf("Got a message %q", *message.Text)
+	if message.Text != nil {
+		log.Printf("Got a message %q", *message.Text)
+	}
+
+	if message.Chat.Type == chatTypePrivate {
+		return s.privateMessage(message)
+	} else {
+		return s.publicChatMessage(message)
+	}
+}
+
+const whoAmI = "I am a bot that can generate a report from your todo list on Github Projects."
+
+func (s *rootConversationState) publicChatMessage(message message) (ConversationStateHandler, []telegramBotActor) {
+	if message.Text == nil {
+		return s, []telegramBotActor{}
+	}
 	switch strings.ToLower(strings.TrimSpace(*message.Text)) {
 	case "/start":
-		return s, []telegramBotActor{sendMessage{
-			ChatID: strconv.FormatInt(message.Chat.ID, 10),
-			Text: `Hi! I am a bot that can generate a report from your todo list on Github Projects.
-
-You can use /help to get a list of commands. The one you will need right now is /addApiKey.`,
-		}}
+		return s, []telegramBotActor{message.sameChatPlain("Hi! " + whoAmI + `
 
+You can use /help to get a list of commands. To get started send me /addApiKey in private messages.`)}
 	case "/help":
-		return s, []telegramBotActor{sendMessage{
-			ChatID: strconv.FormatInt(message.Chat.ID, 10),
-			Text: `Here are the commands I have:
+		return s, []telegramBotActor{message.sameChatPlain(s.helpText())}
+	case "/addapikey", "/listprojects":
+		return s, []telegramBotActor{message.sameChatPlain(`This command only works in private (direct) messages for your privacy and security.`)}
+	default:
+		return s, []telegramBotActor{}
+	}
+}
 
-/help: you are here!
+func (s *rootConversationState) privateMessage(message message) (ConversationStateHandler, []telegramBotActor) {
+	if message.Text == nil {
+		return s, []telegramBotActor{}
+	}
+	switch strings.ToLower(strings.TrimSpace(*message.Text)) {
 
-/addApiKey: Add a GitHub API key`,
-		}}
+	case "/start":
+		return s, []telegramBotActor{message.sameChatPlain("Hi! " + whoAmI + `
 
+You can use /help to get a list of commands. The one you will need right now is /addApiKey`)}
+	case "/help":
+		return s, []telegramBotActor{message.sameChatPlain(s.helpText())}
 	case "/addapikey":
-		return s, []telegramBotActor{sendMessage{
-			ChatID: strconv.FormatInt(message.Chat.ID, 10),
-			Text:   `501: Oops... We are working on it!`,
-		}}
+		return &addApiKeyConversationState{}, []telegramBotActor{message.sameChatMarkdownV2(`Lets set your GitHub API key\. I can only hold one at a time and I will use it to get information about your projects\.
```

Comment:
how about adding all response template messages in one place? We can cover this question on the call.

Reply:
what about this?